src/refactorings/change-signature/change-signature.ts

```diff
@@ -0,0 +1,198 @@
+import { Editor, ErrorReason, SelectedPosition } from "../../editor/editor";
+import { Selection } from "../../editor/selection";
+import * as t from "../../ast";
+import { Path } from "../../editor/path";
+
+let toModifyNode: t.FunctionDeclaration;
+
+export async function changeSignature(editor: Editor) {
+  const params =
+    toModifyNode?.params.map((p, index) => {
```

Comment:
I'm intrigued by the fact we don't seem to parse the code first and we rely on the current value of `toModifyNode` that was set by `createVisitor`. I guess this works because the later is invoked regularly to verify if a change can be made (action provider). 

I need to check what happens if you attempt to execute the refactoring from the command palette instead of the lightbulb. I guess it would still work and it's more performant… but I _feel_ it would be safer to parse the code first to determine the node to modify. Hopefully, that wouldn't be a big change to make.

Reply:
I'm also concerned with the fact the variable persists and what could happen if you trigger the refactoring again from the command palette with an old (maybe invalid) value?